block/manager.go

```diff
@@ -347,8 +356,35 @@ func (m *Manager) IsDAIncluded(hash types.Hash) bool {
 	return m.blockCache.isDAIncluded(hash.String())
 }
 
+// getRemainingSleep calculates the remaining sleep time based on config and a start time.
+func (m *Manager) getRemainingSleep(start time.Time) time.Duration {
+	elapsed := time.Since(start)
+	interval := m.conf.BlockTime
+
+	if m.conf.LazyAggregator {
+		if m.buildingBlock {
+			// LazyBufferTime is used to give time for transactions to
+			// accumulate if we are coming out of a period of inactivity. If we
+			// had recently produced a block (i.e. within the block time) then
+			// we will sleep for the remaining time within the block time
+			// interval.
+			if elapsed >= interval {
+				return m.conf.LazyBufferTime
+			}
+		} else {
+			interval = m.conf.LazyBlockTime
+		}
+	}
```

Comment:
suggesting minor improvements:
```
    if start.After(now) {
        return m.conf.BlockTime
    }

    elapsed := now.Sub(start)
    interval := m.conf.BlockTime

    if m.conf.LazyAggregator {
        if m.buildingBlock && elapsed >= interval {
            return m.conf.LazyBufferTime
        } else if !m.buildingBlock {
            interval = m.conf.LazyBlockTime
        }
    }
    ```

Reply:
What's `now` meaning? Looks like that `start` always before `now`

Reply:
```
if m.buildingBlock && elapsed >= interval {
    return m.conf.LazyBufferTime
} else if !m.buildingBlock {
    interval = m.conf.LazyBlockTime
}
```
this may lower readability

Reply:
Resolving as implemented.